qt/widgets/common/src/MuonFitPropertyBrowser.cpp

```diff
@@ -1063,6 +887,66 @@ bool MuonFitPropertyBrowser::isWorkspaceValid(Workspace_sptr ws) const {
 }
 
 void MuonFitPropertyBrowser::finishHandle(const IAlgorithm *alg) {
+  if (alg->name() == "CalculateMuonAsymmetry") {
+    finishHandleTF(alg);
+  } else {
+    finishHandleNormal(alg);
+  }
+}
+void MuonFitPropertyBrowser::finishHandleTF(const IAlgorithm *alg) {
+
+  // Copy experiment info to output workspace
+  if (AnalysisDataService::Instance().doesExist(outputName() + "_Workspace")) {
+    // Input workspace should be a MatrixWorkspace according to isWorkspaceValid
+    auto inWs = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(
+        static_cast<std::string>(
+            alg->getProperty("UnNormalizedWorkspaceList")));
+    auto outWs = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(
+        outputName() + "_Workspace");
+    if (inWs && outWs) {
+      outWs->copyExperimentInfoFrom(inWs.get());
+    }
+  } else if (AnalysisDataService::Instance().doesExist(outputName() +
+                                                       "_Workspaces")) {
+    // Output workspace was a group
+    auto outGroup = AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>(
+        outputName() + "_Workspaces");
+    if (outGroup->size() == m_workspacesToFit.size()) {
+      for (size_t i = 0; i < outGroup->size(); i++) {
+        auto outWs =
+            boost::dynamic_pointer_cast<MatrixWorkspace>(outGroup->getItem(i));
+        auto inWs = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(
+            m_workspacesToFit[i]);
+        if (inWs && outWs) {
+          outWs->copyExperimentInfoFrom(inWs.get());
+        }
+      }
+    }
+  }
+
+  auto status = QString::fromStdString(alg->getPropertyValue("OutputStatus"));
+  emit fitResultsChanged(status);
+  FitPropertyBrowser::fitResultsChanged(status);
+
+  // If fit was simultaneous, insert extra information into params table
+  // and group the output workspaces
+  const int nWorkspaces = static_cast<int>(m_workspacesToFit.size());
+  if (nWorkspaces > 1) {
+    std::string baseName = outputName();
+    finishAfterTFSimultaneousFit(alg, baseName);
+    getFitResults();
+    std::vector<std::string> wsList =
+        alg->getProperty("UnNormalizedWorkspaceList");
+    emit fittingDone(QString::fromStdString(wsList[0]));
+  } else {
+    getFitResults();
+    std::vector<std::string> wsList =
+        alg->getProperty("UnNormalizedWorkspaceList");
+    emit fittingDone(QString::fromStdString(wsList[0]));
```

Comment:
These 3 lines can be moved out of the `if` and `else` since they are done whether `nWorkspaces > 1` or not.
```
getFitResults();
 std::vector<std::string> wsList =
        alg->getProperty("UnNormalizedWorkspaceList");
emit fittingDone(QString::fromStdString(wsList[0]));
```